src/library_async.js

```diff
@@ -134,7 +137,22 @@ mergeInto(LibraryManager.library, {
               Asyncify.exportCallStack.push(x);
               try {
 #endif
+#if ASYNCIFY == 1 && MEMORY64
+                let args = Array.from(arguments)
+                if (args.length > 0 && x.startsWith('dynCall_')) {
+                    args[0] = Number(args[0])
```

Comment:
Do you know why this is needed.. its seems a little odd, maybe worth a comment?

Reply:
All dynCall_* receive BigInt as a first argument for some reason and it fails with "cannot convert BigInt to a number". I'll check if i can change how dynCall functions are generated

Reply:
The first argument to dynCall will indeed be an i64 (i.e. a pointer), but normally the idea is that when such functions are exported to JS we wrap them in conversion functions so the JS code only deals with i53 numbers instead.